mitmproxy/addons/tutorialapp/static/tutorial.js

```diff
@@ -0,0 +1,76 @@
+var ajaxButtons = document.getElementsByClassName('ajax-btn');
+
+for(let i = 0; i < ajaxButtons.length; i++) {
+    ajaxButtons[i].addEventListener("click", function(e) {
+        fetch(e.currentTarget.href)
+            .then(response => response.json())
+            .then(data => {
+                document.getElementById("votes-cat").innerHTML = data["cat"];
+                document.getElementById("votes-dog").innerHTML = data["dog"];
+            });
+
+        e.preventDefault();
+        return false;
+    })
+}
+
+
+var timer;
+
+function playEvent(e) {
+    console.log("playing at " + this.currentTime);
+    var playStartPosition = this.currentTime;
+    var playStartDate = new Date();
+    var annotations = this.annotations;
+    var instructions = this.instructions;
+
+    timer = setInterval(function(){
+        var now = new Date();
+        updateAnnotations(
+            playStartPosition + (now - playStartDate) / 1000,
```

Comment:
You can just pass `this.currentTime` as first parameter and remove these calculations. No need to manually track time.

Reply:
You are right, that's an unnecessarily complex implementation :) Thanks and sorry, I forgot to reply to this comment.